### PR TITLE
refactor: split caseDocuments fragment methods (FPLA-857)

### PR DESCRIPTION
**Pull request by Jamesnelson117**
**Before creating a pull request make sure that:**

- [x] PR's title includes JIRA ticket number and follow [Conventional Commits](https://www.conventionalcommits.org/) specification
- [x] README and other documentation has been updated / added (if needed)
- [x] tests have been updated / new tests has been added (if needed)

### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/FPLA-857

### Change description ###
Split caseDocument fragment's input of documents / assertion into two methods. 

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: e2e/tests/localAuthorityMaintainsCaseAfterGatekeeping_test.js [hunk 1]
```diff
@@ -36,6 +36,6 @@ Before(async (I, caseViewPage, submitApplicationEventPage, enterFamilyManCaseNum
   await I.navigateToCaseDetails(caseId);
 });
 
-Scenario('local authority uploads documents', uploadDocs.uploadDocuments());
+Scenario('local authority uploads documents', uploadDocs.assertMandatoryDocuments(), uploadDocs.uploadMandatoryDocuments());
```
Comment (lewisbirks) [inline e2e/tests/localAuthorityMaintainsCaseAfterGatekeeping_test.js:39]: Looks like it is in the wrong order, would it be possible to maybe change it to read something like
```js
Scenario('local authority uploads documents', async (I, caseViewPage, uploadDocumentsEventPage) => {
  // navigate to event...
  uploadDocs.uploadMandatoryDocuments()
  // complete event...
  // navigate to tab...
  uploadDocs.assertMandatoryDocuments()
});
```

Comment (Jamesnelson117) [reply e2e/tests/localAuthorityMaintainsCaseAfterGatekeeping_test.js:39]: Works right to left. Yeah was not sure wether we wanted to stick to the one line methods. I'll expand them out.

File: e2e/fragments/caseDocuments.js [hunk 2]
```diff
@@ -1,62 +1,42 @@
+const { I } = inject();
 const config = require('../config.js');
 
 module.exports = {
-  uploadDocuments() {
-    return async (I, caseViewPage, uploadDocumentsEventPage) => {
-      await caseViewPage.goToNewActions(config.applicationActions.uploadDocuments);
-      uploadDocumentsEventPage.selectSocialWorkChronologyToFollow(config.testFile);
+  uploadMandatoryDocuments(uploadDocumentsEventPage, uploadSWET=true) {
```
Comment (tomaszpowroznik) [inline e2e/fragments/caseDocuments.js:5]: are these documents really mandatory?

File: e2e/tests/localAuthorityMaintainsCaseAfterGatekeeping_test.js [hunk 3]
```diff
@@ -36,6 +36,21 @@ Before(async (I, caseViewPage, submitApplicationEventPage, enterFamilyManCaseNum
...
+  await I.completeEvent('Save and continue');
+  I.seeEventSubmissionConfirmation(config.applicationActions.uploadDocuments);
+  caseViewPage.selectTab(caseViewPage.tabs.documents);
+  uploadDocs.assertMandatoryDocuments();
+});
 
-Scenario('local authority uploads court bundle', uploadDocs.uploadCourtBundle());
+Scenario('local authority uploads court bundle', async (I, caseViewPage, uploadDocumentsEventPage) => {
+  await caseViewPage.goToNewActions(config.applicationActions.uploadDocuments);
+  I.seeElement(uploadDocumentsEventPage.documents.courtBundle);
```
Comment (tomaszpowroznik) [inline e2e/tests/localAuthorityMaintainsCaseAfterGatekeeping_test.js:50]: maybe it should be part of uploadDocumentsEventPage.uploadCourtBundle

File: e2e/pages/events/uploadDocumentsEvent.page.js [hunk 4]
```diff
@@ -101,4 +101,42 @@ module.exports = {
...
+      this.uploadSWET(config.testFile);
+    } else {
+      this.selectSWETAsNotRequired();
+    }
+
+    this.uploadThresholdDocument(config.testFile);
+    this.uploadChecklistDocument(config.testFile);
+  },
+
+  assertCaseDocuments(hasUploadedSWET=true) {
```
Comment (tomaszpowroznik) [inline e2e/pages/events/uploadDocumentsEvent.page.js:127]: the thing is that page object is intended to represent particular page, assertions are done on different page (tab)

Comment (Jamesnelson117) [reply e2e/pages/events/uploadDocumentsEvent.page.js]: True. I've changed caseDocuments fragment to a helper class as helpers offer consistent functionality as to what we want from our current caseDocuments fragment. Reverted using a page object method to enter all case documents as I think this is better suited for the helper.

File: e2e/tests/localAuthorityMaintainsCaseAfterGatekeeping_test.js (hunk 3 above)
Comment (Jamesnelson117) [reply e2e/tests/localAuthorityMaintainsCaseAfterGatekeeping_test.js:50]: Page method should just upload document in my opinion.

**Pull request merged**